kani-compiler/src/kani_middle/transform/body.rs

```diff
@@ -146,9 +150,10 @@ impl MutableBody {
     /// Add a new assert to the basic block indicated by the given index.
     ///
     /// The new assertion will have the same span as the source instruction, and the basic block
-    /// will be split. The source instruction will be adjusted to point to the first instruction in
-    /// the new basic block.
-    pub fn add_check(
+    /// will be split. If `InsertPosition` is `InsertPosition::Before`, `source` will point to the
+    /// same instruction as before. If `InsertPosition` is `InsertPosition::After`, `source` will
+    /// point to the new terminator.
+    pub fn new_check(
```

Comment:
Can you please explain why `new` makes more sense here? For me, `new` implies that this method is returning a new object, not necessarily it's adding to the body. Another option would be to use `insert_` instead.

I acknowledge that not all methods are consistent with this terminology, but I would prefer fixing that. E.g.: `new_assignment` could become `add_assignment` / `insert_assignment`.

Reply:
I am happy to change the names of all methods that "add" constructs to the body to "insert_XXX" to be consistent